# Ejemplo 3.4.1 de Díaz y Morales (2015)
<img src="https://raw.githubusercontent.com/fhernanb/fhernanb.github.io/master/my_docs/logo_unal_color.png" alt="drawing" width="200"/>

En este ejemplo se desea estudiar el siguiente conjunto de hipótesis usando un nivel de significancia de 0.05.

$H_0: \boldsymbol{\mu} = (70, 170)^\top$

$H_A: \boldsymbol{\mu} \neq (70, 170)^\top$

asumiendo que la matriz de varianzas es conocida así:

\begin{equation}
\boldsymbol{\Sigma} = 
\begin{pmatrix}
20 & 100 \\
100 & 1000
\end{pmatrix}
\end{equation}

In [1]:
import numpy as np
import pandas as pd
import scipy.stats

A continuación los datos del ejemplo

In [2]:
dt = {'x1': [69, 74, 68, 70, 72, 67, 66, 70, 76, 68,
             72, 79, 74, 67, 66, 71, 74, 75, 75, 76], 
      'x2': [153, 175, 155, 135, 172, 150, 115, 137, 200, 130,
             140, 265, 185, 112, 140, 150, 165, 185, 210, 220]}
dt = pd.DataFrame(data=dt)
dt

x1   x2
0   69  153
1   74  175
2   68  155
3   70  135
4   72  172
5   67  150
6   66  115
7   70  137
8   76  200
9   68  130
10  72  140
11  79  265
12  74  185
13  67  112
14  66  140
15  71  150
16  74  165
17  75  185
18  75  210
19  76  220

<div class="alert alert-danger">
  <strong>Cuidado!</strong> Se debe verificar que la muestra aleatoria provenga de una población normal multivariada. Aquí vamos a asumir que este supuesto se cumple.
</div>

El vector de medias **muestral** se obtiene así:

In [3]:
x_barra = dt.mean()
x_barra = np.matrix(x_barra).transpose()  # Para convertir a matriz columna
print(x_barra)                            # Para ver el contenido

[[ 71.45]
 [164.7 ]]


La matriz de varianzas **poblacional** conocida se construye así:

In [4]:
Sigma = np.matrix([[20, 100],
                   [100, 1000]])
Sigma

matrix([[  20,  100],
        [ 100, 1000]])

El vector de referencia es $\boldsymbol{\mu}_0=(70, 170)^\top$.

In [5]:
mu0 = [[70], [170]]
mu0

[[70], [170]]

Ahora calculamos $T^2=n (\bar{\boldsymbol{x}}-\boldsymbol{\mu}_0)^\top \boldsymbol{\Sigma}^{-1} (\bar{\boldsymbol{x}}-\boldsymbol{\mu}_0)$, la cantidad $(\bar{\boldsymbol{x}}-\boldsymbol{\mu}_0)$ se va a llamar `aux` para facilitar las operaciones.

In [6]:
n = 20 # número de observaciones
aux = np.matrix((x_barra - mu0)) # Elemento auxiliar
T2 = n * np.matmul(np.matmul(aux.transpose(), Sigma.I), aux)
T2.item()  # Para obtener el T2 como un escalar y no como una matriz

8.402600000000033

El valor crítico para realizar la prueba es $\chi_{p}(\alpha)$. Para obtener el valor crítico con $\alpha=0.05$ se procede así:

In [7]:
p = 2 # Número de variables en el problema
scipy.stats.chi2.ppf(q=1-0.05, df=p)  # Valor crítico

5.991464547107979

<div class="alert alert-success">
  <strong>Resultado</strong> Como el estadístico es $T^2=8.40$, que es mayor que el valor crítico $5.99$, entonces hay evidencias suficientes para rechazar $H_0: \boldsymbol{\mu} = (70, 170)^\top$.
</div>

# Análisis marginal

En esta sección vamos a construir intervalos de confianza para ambas medias pero de forma separada.

Para construir el intervalo de confianza del 95% para $\mu_1$ se procede así:

In [8]:
x1_barra = x_barra[0].item()
sigma1 = 20 ** 0.5 # Desviación, no varianza
n1 = 20
z = scipy.stats.norm.ppf(1-0.025, loc=0, scale=1) # Cuantil 0.975 de N(0, 1)

linf = x1_barra - z * sigma1 / np.sqrt(n1)
lsup = x1_barra + z * sigma1 / np.sqrt(n1)

print("El límite inferior es", linf, " y el límite superior es", lsup)

El límite inferior es 69.49003601545995  y el límite superior es 73.40996398454006


Como $\mu_{10}=70$ está entre los límites del intervalo, no hay evidencias para rechazar $H_0: \mu_1=70$.

Para construir el intervalo de confianza del 95% para $\mu_2$ se procede así:

In [9]:
x2_barra = x_barra[1].item()
sigma2 = 1000 ** 0.5 # Desviación, no varianza
n2 = 20
z = scipy.stats.norm.ppf(1-0.025, loc=0, scale=1) # Cuantil 0.975 de N(0, 1)

linf = x2_barra - z * sigma2 / np.sqrt(n2)
lsup = x2_barra + z * sigma2 / np.sqrt(n2)

print("El límite inferior es", linf, " y el límite superior es", lsup)

El límite inferior es 150.84096175650322  y el límite superior es 178.55903824349676


Como $\mu_{20}=170$ está entre los límites del intervalo, no hay evidencias para rechazar $H_0: \mu_2=170$.